UnitySDK/Assets/ML-Agents/Scripts/Sensor/StackingSensor.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections.Generic;
+using UnityEngine.Assertions.Comparers;
+
+namespace MLAgents.Sensor
+{
+    /// <summary>
+    /// Sensor that wraps around another Sensor to provide temporal stacking.
+    /// </summary>
+    public class StackingSensor : SensorBase
+    {
+        /// <summary>
+        /// The wrapped sensor.
+        /// </summary>
+        SensorBase m_WrappedSensor;
+
+        /// <summary>
+        /// Number of stacks to save
+        /// </summary>
+        int m_NumStackedObservations;
+        int m_NumUnstackedObservationSize;
+
+        string m_Name;
+        int[] m_Shape;
+
+        /// <summary>
+        /// Buffer of previous observations
+        /// </summary>
+        // TODO Flat list like old code? Circular buffer?
+        List<float> m_StackedObservations;
```

Comment:
Note that this is like the current implementation for now. I think it would be cleaner to have a `List<float[]>` and use that like a circular buffer instead.

Reply:
could you go even further and just make it a flat `float[]`?

Reply:
Uses a float[][] now.